tests/queryset_pickle/tests.py

```diff
@@ -153,3 +153,27 @@ def test_unsupported_unpickle(self):
         msg = "Pickled queryset instance's Django version 1.0 does not match the current version %s." % get_version()
         with self.assertRaisesMessage(RuntimeWarning, msg):
             pickle.loads(pickle.dumps(qs))
+
+    def test_queryset_evaluation(self):
+        """
+        Pickling a QuerySet.query with an __in=inner_qs lookup,
+        shouldn't evaluate inner_qs.
+        """
+        for i in range(1, 5):
+            Group(name='Group {}'.format(i)).save()
+
+        Event(title='Event 1', group=Group.objects.all()[0]).save()
+
+        groups = Group.objects.all()
+        events = Event.objects.filter(group__in=groups)
+
+        with self.assertNumQueries(0):
+            dumped = pickle.dumps(events.query)
+
+        # Make sure that unpickle doesn't cause evaluation
+        with self.assertNumQueries(0):
+            reloaded = pickle.loads(dumped)
+            reloaded_events = Event.objects.none()
+            reloaded_events.query = reloaded
+
+        self.assertQuerysetEqual(events, reloaded_events, lambda x: x)
```

Comment:
Where in the process of replacing these constructs with `self.assertSequenceEqual`, see #7226.